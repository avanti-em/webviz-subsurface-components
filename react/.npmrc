engine-strict=true
ignore-scripts=true
CYPRESS_INSTALL_BINARY=0