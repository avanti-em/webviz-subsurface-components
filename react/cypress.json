{
    "baseUrl": "http://localhost:6006/?path=/story/deckglmap--default",
    "video": true,
    "screenshotsFolder": "./cypress/snapshots/actual",
    "trashAssetsBeforeRuns": true,
    "env": {
        "failSilently": false,
        "ALWAYS_GENERATE_DIFF": false,
        "type": "base"
    }
}
