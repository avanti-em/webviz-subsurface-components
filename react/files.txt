cypress-visual-screenshots/comparison/defaultMap.cy-default-map-story.png
cypress-visual-screenshots/diff/defaultMap.cy-default-map-story.png
files.txt
