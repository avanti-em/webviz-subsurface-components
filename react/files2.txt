cypress-visual-screenshots/diff/defaultMap.cy-default-map-story.png
