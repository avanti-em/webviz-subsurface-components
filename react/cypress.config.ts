import { defineConfig } from "cypress";
const getCompareSnapshotsPlugin = require("cypress-image-diff-js/dist/plugin");

export default defineConfig({
  component: {
    devServer: {
      framework: "create-react-app",
      bundler: "webpack",
    },
    video:false,
    setupNodeEvents(on, config) {
      getCompareSnapshotsPlugin(on, config);
    }
  },
});
