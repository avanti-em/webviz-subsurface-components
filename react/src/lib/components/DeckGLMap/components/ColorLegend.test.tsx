/* eslint-disable @typescript-eslint/no-var-requires */
import { render } from "@testing-library/react";
import "jest-styled-components";
import "@testing-library/jest-dom";
import React from "react";
import ColorLegend from "./ColorLegend";
import { Layer } from "deck.gl";
import {
    ExtendedLayer,
    getLayersInViewport,
    getLayersWithDefaultProps,
} from "../layers/utils/layerTools";

const exampleData = require("../../../../demo/example-data/deckgl-map.json");
const layers = getLayersWithDefaultProps(exampleData[0].layers);
const colorTables = require("@emerson-eps/color-tables/dist/component/color-tables.json");

describe("Test Color Legend component", () => {
    it("snapshot test", () => {
        const { container } = render(
            <ColorLegend
                layer={[] as unknown as ExtendedLayer<unknown>}
                colorTables={colorTables}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
