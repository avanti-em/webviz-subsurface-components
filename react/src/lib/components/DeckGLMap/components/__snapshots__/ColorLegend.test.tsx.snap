// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Color Legend component snapshot test 1`] = `null`;
